crates/runtime/src/lib.rs

```diff
@@ -304,14 +333,18 @@ impl Runtime {
     }
 
     pub async fn start_pods_watcher(&mut self) -> notify::Result<()> {
-        let mut current_app = Arc::clone(&self.app);
-
         let mut rx = self.pods_watcher.watch()?;
 
         while let Some(new_app) = rx.recv().await {
+            let current_app = self.app.read().await;
+            let current_datasets = current_app.datasets.clone();
+            let current_models = current_app.models.clone();
+
             tracing::debug!("Updated pods information: {:?}", new_app);
             tracing::debug!("Previous pods information: {:?}", current_app);
 
+            drop(current_app);
+
             let mut auth = auth::AuthProviders::default();
```

Comment:
Let's do the same thing with auth. Move it behind an Arc<RwLock<>> and then just update the auth here so auth is updated throughout the runtime instead of just passing it in to the `load()` methods.